packages/validation/src/validateField/resolveFieldRestrictions.ts

```diff
@@ -35,64 +67,5 @@ export const resolveFieldRestrictions = (
 		return [];
 	}
 
-	switch (field.valueType) {
-		case 'boolean': {
-			const output: FieldRestrictionRule[] = [];
-			if (field.restrictions.required) {
-				output.push({ type: 'required', rule: field.restrictions.required });
-			}
-			// if (field.restrictions.script) {
-			// 	output.push({ type: 'script', rule: asArray(field.restrictions.script) });
-			// }
-			return output;
-		}
-		case 'integer': {
-			const output: FieldRestrictionRule[] = [];
-			if (Array.isArray(field.restrictions.codeList)) {
-				output.push({ type: 'codeList', rule: field.restrictions.codeList });
-			}
-			if (field.restrictions.range) {
-				output.push({ type: 'range', rule: field.restrictions.range });
-			}
-			if (field.restrictions.required) {
-				output.push({ type: 'required', rule: field.restrictions.required });
-			}
-			// if (field.restrictions.script) {
-			// 	output.push({ type: 'script', rule: asArray(field.restrictions.script) });
-			// }
-			return output;
-		}
-		case 'number': {
-			const output: FieldRestrictionRule[] = [];
-			if (Array.isArray(field.restrictions.codeList)) {
-				output.push({ type: 'codeList', rule: field.restrictions.codeList });
-			}
-			if (field.restrictions.range) {
-				output.push({ type: 'range', rule: field.restrictions.range });
-			}
-			if (field.restrictions.required) {
-				output.push({ type: 'required', rule: field.restrictions.required });
-			}
-			// if (field.restrictions.script) {
-			// 	output.push({ type: 'script', rule: asArray(field.restrictions.script) });
-			// }
-			return output;
-		}
-		case 'string': {
-			const output: FieldRestrictionRule[] = [];
-			if (Array.isArray(field.restrictions.codeList)) {
-				output.push({ type: 'codeList', rule: field.restrictions.codeList });
-			}
-			if (field.restrictions.regex) {
-				output.push({ type: 'regex', rule: field.restrictions.regex });
-			}
-			if (field.restrictions.required) {
-				output.push({ type: 'required', rule: field.restrictions.required });
-			}
-			// if (field.restrictions.script) {
-			// 	output.push({ type: 'script', rule: asArray(field.restrictions.script) });
-			// }
-			return output;
-		}
-	}
+	return TypeUtils.asArray(field.restrictions).flatMap(extractRulesFromRestrictionObject);
```

Comment:
Convert restrictions into an array if it was a single object, simplifies processing to always be done the same way.

Reply:
woohoo native flatmap